Add tag system001

MoltndarKen opened the pull request:
# WHAT
## introduce 'acts-as-taggable-on' to implement tag systems
- introduct gem 'acts-as-taggable-on'
- relate between prototype and tags
- generate tags controller
- edit view files to save tag 
- edit prototype controller to save datas including tags
# WHY
## implement tag systems
# To do the next
- add tag_ index file and relate with users
-  screen transition from the tags to prototypes

Commit:
# WHAT& WHY
run
'bin/bundle
bin/rake acts_as_taggable_on_engine:install:migrations
bin/rake db:migrate' to make tag systems by acts-as-taggable-on

Commit:
# WHAT&WHY
add  'acts_as_taggable' to associate between tag and prototype

Commit:
# WHAT&WHY
improve sass:error

Commit:
# what&why
edit prototype controller to post 'tag'

Commit:
# WHAT&WHY
add new,show view files about prototypes and configure routing

Commit:
# WHAT&WHY
add tag_list to prototype.rb

Commit:
# WHAT&WHY
replace new notification about hash

Commit:
# WHAT&WHY
set the name property to form tag

Commit:
# what&why
not generate test,helper,assets files automatically and generate tags controller

Commit:
# what&why
generate catch copy, concept columns to Prototypes tables

Commit:
# what&why
remove 'introduction' column from prototypes table

Commit:
# WHAT
replace form_for notification about prototype/new view file
# WHY
escape from error because of using form elements

Commit:
# what&why
can save datas

Commit:
# what& why
delete unnecesarry files

Commit:
# what & why
To prevent the changes that you do not want when you merge

Commit:
# what&why
replace new notification about hash

Commit:
# what&why
replace new notification about hash

Commit:
# what&why
place a blank line

Commit:
# what&why
place a blank line in the end

Commit:
# what
installing gem 'jquery-ui-rails'
# why
to introduce Tag-it

ogqwa commented on Gemfile:
```diff
@@ -44,8 +44,10 @@ gem 'devise'
+gem 'carrierwave'
+gem 'html2haml'
+gem 'rmagick'
+gem 'jquery-ui-rails'
```
空行を入れましょう

ogqwa commented on app/models/prototype.rb:
```diff
@@ -1,2 +1,4 @@
 class Prototype < ActiveRecord::Base
+  acts_as_taggable
+  acts_as_taggable_on :web_design,:user_interface,:application_about
```
これだとタグの名前が限定的なものになってしまうので、:listのような名前で管理しましょう

ogqwa commented on app/views/layouts/application.html.haml:
```diff
@@ -1,12 +1,14 @@
+    .dower
+      = yield
     = render "layouts/footer"
+    
```
空行を入れましょう

ogqwa replied:
こちら修正しましたか？

MoltndarKen replied:
修正が反映されていませんでした、お手数かけました💦

ogqwa commented on config/routes.rb:
```diff
@@ -1,4 +1,6 @@
 Rails.application.routes.draw do
+  get 'tags/index'
```
tagはタグの一覧を表示するindexページ以外に、そのタグを持つprototypeの一覧を表示するshowページもあるのでresourcesでshowアクションも設定しましょう

Commit:
# what&why
improve tags'names and place blank line, and add show action to routing about tag.

ogqwa commented on app/models/prototype.rb:
```diff
@@ -1,2 +1,4 @@
 class Prototype < ActiveRecord::Base
+  acts_as_taggable
+  acts_as_taggable_on :tag1, :tag2, :tag3
```
acts_as_taggable_on :tag1, :tag2, :tag3ではなく
acts_as_taggable_on :listのような定義の仕方で保存できるようにしましょう！

MoltndarKen replied:
> acts_as_taggable_on :lis
> コントローラー、ビューファイルにprototype投稿画面で:listを用いて複数保存した場合、どのように記述するといいでしょうか？？投稿画面には3つタグを保存できるようにしてあり、同じ:listを使うと、保存時に３つ目のタグの内容のみが保存されるようになってしまいます。。

MoltndarKen replied:
解決しました!稲垣くんのqiitaの投稿を参考にしました。

[act-as-taggable-onを使ってタグを複数投稿する](http://qiita.com/sinagaki58/items/b550cc624a75022ba68f)

Commit:
# what& why
place a blank line

Commit:
# what&why
can save multiple tags

Commit:
Merge branch 'master' into add_tag_system001

Conflicts:
	Gemfile
	app/controllers/prototypes_controller.rb
	app/models/prototype.rb
	config/routes.rb
